dedupe/core.py

```diff
@@ -71,6 +71,8 @@ def randomPairsMatch(n_records_A, n_records_B, sample_size):
 
     if sample_size >= n:
         random_pairs = numpy.arange(n)
+    elif n > numpy.iinfo(int).max:
+        random_pairs = numpy.array(random.sample(range(n), sample_size))
```

Comment:
`i, j` need to be ints when this function returns. Will they be. 

If numpy.unravel_index does the right thing with floats, then we could simply this conditional by never doing a `dtype=int`